File: app/src/main/java/com/OdiousPanda/rainbow/DAOs/CoordinateDAO.java
package com.OdiousPanda.rainbow.DAOs;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.OdiousPanda.rainbow.DataModel.Coordinate;

import java.util.List;

@Dao
public interface CoordinateDAO {
    @Insert
    void insert(Coordinate coordinate);

    @Update
    void update(Coordinate coordinate);

    @Delete
    void delete(Coordinate coordinate);

    @Query("Select * from location_table")
    LiveData<List<Coordinate>> selectAll();
}

File: app/src/main/java/com/odiousPanda/rainbowKt/database/CoordinateDAO.kt
package com.odiousPanda.rainbowKt.database

import androidx.lifecycle.LiveData
import androidx.room.*
import com.odiousPanda.rainbowKt.model.dataSource.Coordinate

@Dao
interface CoordinateDAO {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(coordinate: Coordinate)

    @Update
    fun update(coordinate: Coordinate)

    @Delete
    fun delete(coordinate: Coordinate)

    @Query("Select * from location_table")
    fun selectAll(): LiveData<List<Coordinate>>
}
